packages/database_v2/artifacts/init.sql

```diff
@@ -0,0 +1,168 @@
+DROP TYPE IF EXISTS resource_status CASCADE;
+CREATE TYPE resource_status AS ENUM (
+    'Queued',
+    'URIParseFailed',
+    'ContentFetchFailed',
+    'PinRequestFailed',
+    'ContentLinked'
+);
+
+DROP TYPE IF EXISTS nft_asset_status CASCADE;
+CREATE TYPE nft_asset_status AS ENUM (
+    'Queued',
+    'URIParseFailed',
+    'ContentFetchFailed',
+    'ContentParseFailed',
+    'PinRequestFailed',
+    'Linked'
+);
+
+DROP TYPE IF EXISTS pin_status CASCADE;
+CREATE TYPE pin_status AS ENUM (
+    'PinError',
+    'PinQueued',
+    'Pinned',
+    'Pinning'
+);
+
+DROP TYPE IF EXISTS pin_service CASCADE;
+CREATE TYPE pin_service AS ENUM (
+    'Pinata',
+    'IpfsCluster'
+);
+
+CREATE TABLE IF NOT EXISTS blockchain_block (
+    hash TEXT NOT NULL,
+    number INTEGER NOT NULL,
+
+    updated_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    inserted_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    PRIMARY KEY (hash)
+);
+
+
+CREATE TABLE IF NOT EXISTS nft (
+    id TEXT NOT NULL,
+    token_id TEXT NOT NULL,
+    mint_time TIMESTAMP(3) NOT NULL,
+    token_uri TEXT NOT NULL,
+    contract_id TEXT NOT NULL,
+    nft_owner_id TEXT NOT NULL,
+
+    updated_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    inserted_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS nfts_by_blockchain_blocks (
+    blockchain_block_hash TEXT NOT NULL,
+    nft_id TEXT NOT NULL,
+
+    updated_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    inserted_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    PRIMARY KEY (blockchain_block_hash, nft_id)
+);
+
+CREATE TABLE IF NOT EXISTS nft_owner (
+    id TEXT NOT NULL,
+
+    updated_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    inserted_at TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS nft_asset (
+    token_uri TEXT NOT NULL,
+    ipfs_URL TEXT,
+    status nft_asset_status NOT NULL,
+    status_text TEXT NOT NULL,
```

Comment:
We have a problem here:

`nft` maps to `nft_asset` by `token_uri` but it can not map to `nft_metadat` by `token_uri` because different `token_uri`s may resolve to the same `content_cid`.

To address this we should have nullable `content_cid` in `nft_asset` table and we should remove `token_uri` from the `nft_metadata` table.

Reply:
Thinking more about it I am wondering if we should remove `nft_asset` table all together because:

1. It is (almost) identical to `resource` table and was mostly different in fauna so they would map to different GraphQL types.
2. Number of times I wanted to drop `nft_asset` in favor of `resource` type however in GraphQL it did not align because `TokenAsset` had ref link to `Metadata` corresponding to it, while `Resource` did not had one (at least semantically it meant different thing).

Reply:
Also worth noting that `status` field for `nft_asset` is larger enum because we do parse it while in case of resource we do not.